core/src/cluster_info.rs

```diff
@@ -1541,11 +1541,17 @@ impl ClusterInfo {
         self.stats
             .new_pull_requests_count
             .add_relaxed(pulls.len() as u64);
+        let peers: Vec<Pubkey> = pulls.iter().map(|(peer, _, _, _)| *peer).dedup().collect();
```

Comment:
good point! How about a comment/assert that there should only be <= 2 peers (entrypoint + randomly selected pull peer), for future reference?

Reply:
added comment